src/ssmDocument/commands/openDocumentItem.ts

```diff
@@ -0,0 +1,134 @@
+        })
+        vscode.window.showTextDocument(textDocument)
+    } catch (err) {
+        result = 'Failed'
+        const error = err as Error
+        vscode.window.showErrorMessage(
+            localize(
+                'AWS.message.error.ssmDocumet.openDocument.could_not_open',
+                'Could not fetch and display document {0} contents',
+                node.documentName
+            )
+        )
+        logger.error('Error on openning document: %0', error)
+    } finally {
+        telemetry.recordSsmOpenDocument({ result: result })
+    }
+}
+
+async function promptUserforDocumentFormat(formats: string[]): Promise<string | undefined> {
+    // Prompt user to pick document format
+    const quickPickItems: SsmDocumentQuickPickItem[] = formats.map(format => {
+        return {
+            label: format,
+            description: `Open document with format ${format}`,
+        }
+    })
+
+    const formatPick = picker.createQuickPick({
+        options: {
+            ignoreFocusOut: true,
+            title: localize('AWS.message.prompt.selectSsmDocumentFormat.placeholder', 'Select a document format'),
+        },
+        buttons: [vscode.QuickInputButtons.Back],
+        items: quickPickItems,
+    })
+
+    const formatChoices = await picker.promptUser({
+        picker: formatPick,
+        onDidTriggerButton: (_, resolve) => {
+            resolve(undefined)
+        },
+    })
+
+    const formatSelection = picker.verifySinglePickerOutput(formatChoices)
+
+    // User pressed escape and didn't select a template
+    if (formatSelection === undefined) {
+        return undefined
+    }
+
+    return formatSelection.label
+}
+
+async function promptUserforDocumentVersion(versions: SSM.Types.DocumentVersionInfo[]): Promise<string | undefined> {
+    // Prompt user to pick document version
+    const quickPickItems: SsmDocumentQuickPickItem[] = []
+    versions.forEach(version => {
+        if (version.DocumentVersion) {
+            quickPickItems.push({
+                label: version.DocumentVersion,
+                description: `${version.IsDefaultVersion ? 'Default' : ''}`,
+            })
+        }
+    })
+
+    if (quickPickItems.length > 1) {
+        const versionPick = picker.createQuickPick({
+            options: {
+                ignoreFocusOut: true,
+                title: localize('AWS.message.prompt.selectSsmDocumentVersion.placeholder', 'Select a document version'),
+            },
+            buttons: [vscode.QuickInputButtons.Back],
+            items: quickPickItems,
+        })
+
+        const versionChoices = await picker.promptUser({
+            picker: versionPick,
+            onDidTriggerButton: (_, resolve) => {
+                resolve(undefined)
+            },
+        })
+
+        const versionSelection = picker.verifySinglePickerOutput(versionChoices)
+
+        // User pressed escape and didn't select a template
```

Comment:
Should this scenario be allowed? If so, I would recommend instead having a quick pick item to select an undefined version explicitly. It's jarring here because it looks like `undefined` is a viable input if the user doesn't enter [this `if` statement](https://github.com/aws/aws-toolkit-vscode/pull/1288/files#diff-6c1ce33f48f364a936e2aa7e9e9c14bfR30).

Reply:
I don't believe we can prevent users from pressing escape on a picker or input. The API returns undefined in those cases.

Reply:
That's fair. This `if` statement isn't necessary in that case, just change the return statement to `return versionSelection?.label` instead.

Reply:
Will remove if statement and change to suggested return statement